Mu2eG4/src/constructExtMonFNALDetector.cc

```diff
@@ -0,0 +1,814 @@
+                                      hlen,
+                                      vacuumMaterial,
+                                      planeRot,
+                                      centerInRoom,
+                                      mother,
+                                      vdId,
+                                      vdIsVisible,
+                                      G4Color::Cyan(),
+                                      vdIsSolid,
+                                      forceAuxEdgeVisible,
+                                      placePV,
+                                      doSurfaceCheck
+                                      );
+
+          // vd are very thin, a more thorough check is needed
+          if (doSurfaceCheck) {
+            checkForOverlaps( vdInfo.physical, config, verbosityLevel>0);
+          }
+        }
+      } // for(vdId-1)
+    } // detector VD block
+
+  }
+
+  //================================================================
+  void constructExtMonFNALModules(const VolumeInfo& mother,
+                                  const G4ThreeVector& offset,
+                                  unsigned iplane,
+                                  const ExtMonFNALModule& module,
+                                  const ExtMonFNALPlaneStack& stack,
+                                  const std::string& volNameSuffix,
+                                  const SimpleConfig& config,
+                                  bool const forceAuxEdgeVisible,
+                                  bool const doSurfaceCheck,
+                                  bool const placePV
+                                  )
+  {
+
+    GeomHandle<ExtMonFNAL::ExtMon> extmon;
+    AntiLeakRegistry& reg = art::ServiceHandle<Mu2eG4Helper>()->antiLeakRegistry();
+    const auto geomOptions = art::ServiceHandle<GeometryService>()->geomOptions();
+    geomOptions->loadEntry( config, "extMonFNALModule", "extMonFNAL.module" );
+    bool const isModuleVisible = geomOptions->isVisible("extMonFNALModule");
+    bool const isModuleSolid   = geomOptions->isSolid("extMonFNALModule");
+
+    unsigned nmodules = stack.planes()[iplane].module_zoffset().size();
+
+    for(unsigned imodule = 0; imodule < nmodules; ++imodule) {
+
+      std::ostringstream osm;
+      osm<<"EMFModule"<<volNameSuffix<<iplane<<imodule;
+
+      CLHEP::HepRotation *stackRotationInMotherInv =
+      reg.add(stack.rotationInMu2e().inverse() * extmon->detectorMotherRotationInMu2e());
+      CLHEP::HepRotation* stackRotationInMother = reg.add(stackRotationInMotherInv->inverse());
+
+
+      bool stackRotation = config.getBool("extMonFNAL.stackRotation");
+      CLHEP::HepRotation* mRotInPlane = reg.add(new G4RotationMatrix);
+
+      mRotInPlane->rotateZ(stack.planes()[iplane].module_rotation()[imodule]);
+
+      if( stack.planes()[iplane].module_zoffset()[imodule] < 0.0 ) {
+        mRotInPlane->rotateY(M_PI);
+      }
+
+      CLHEP::Hep3Vector soffset;
+      CLHEP::HepRotation* mRot;
```

Comment:
Please initialize the pointer.   Perhaps just say "mRot = nullptr", then the rest of the code can be left as is.   The alternative would be to use "mRot = stackRotation ? A : B" construct.